docs/core/tools/dotnet-environment-variables.md

```diff
@@ -1,114 +1,325 @@
+    }
+    ```
+
+1. Setting environment variable value `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` to `true` or `1`.
+
+> [!IMPORTANT]
+> A value set in the project file or _runtimeconfig.json_ has a higher priority than the environment variable.
+
+For more information, see [.NET Globalization Invariant Mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md).
+
+#### `DOTNET_SYSTEM_GLOBALIZATION_USENLS`
+
+This applies to Windows only, for Globalization to use the National Language Support (NLS), set `DOTNET_SYSTEM_GLOBALIZATION_USENLS` to either `true` or `1`. To not use it, set `DOTNET_SYSTEM_GLOBALIZATION_USENLS` to either `false` or `0`.
+
+### `DOTNET_SYSTEM_NET_SOCKETS_*`
+
+This section focuses on two `System.Net.Sockets` environment variables:
+
+- `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS`
+- `DOTNET_SYSTEM_NET_SOCKETS_THREAD_COUNT`
+
+Socket continuations are dispatched to the <xref:System.Threading.ThreadPool?displayProperty=fullName> from the event thread. This avoids continuations blocking the event handling. To allow continuations to run directly on the event thread, set `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS` to `1`. It's disabled by default.
+
+> [!NOTE]
+> This setting can make performance worse if there is expensive work that will end up holding onto the IO thread for longer than needed. Test to make sure this setting helps performance.
+
+Using TechEmpower benchmarks that generate a lot of small socket reads and writes under a very high load, it was observed that a single socket engine is capable of keeping busy up to thirty x64 and eight ARM64 CPU Cores. The vast majority of real-life scenarios will never generate such a huge load (hundreds of thousands of requests per second)
+and having a single producer should be almost always enough. However, to be sure that extreme loads can be handled the `DOTNET_SYSTEM_NET_SOCKETS_THREAD_COUNT` can be used to override the calculated value. When not overridden, the following value is used:
+
+- When `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS` is `1`, the <xref:System.Environment.ProcessorCount?displayProperty=nameWithType> value is used.
+- When `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS` is not `1`, <xref:System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture?displayProperty=nameWithType> is evaluated:
+  - When ARM or ARM64 the cores per engine are set to `8`, otherwise `30`.
+- Using the determined cores per engine, the maximum value of either `1` or <xref:System.Environment.ProcessorCount?displayProperty=nameWithType> over the cores per engine.
+
+#### `DOTNET_SYSTEM_NET_DISABLEIPV6`
+
+The `DOTNET_SYSTEM_NET_DISABLEIPV6` environment variable is used to help determine whether or not Internet Protocol version 6 (IPv6) is disabled. When `DOTNET_SYSTEM_NET_DISABLEIPV6` is set to either `true` or `1`, IPv6 is disabled unless otherwise specified in the <xref:System.AppContext?displayProperty=nameWithType>.
```

Comment:
```suggestion
Helps determine whether or not Internet Protocol version 6 (IPv6) is disabled. When set to either `true` or `1`, IPv6 is disabled unless otherwise specified in the <xref:System.AppContext?displayProperty=nameWithType>.
```